docs/source/nodes_and_pipelines/namespaces.md

```diff
@@ -0,0 +1,336 @@
+```
+
+## Example: Combining disconnected pipelines
+
+Sometimes two pipelines must be connected, but do not share any catalog dependencies. In this example, there is a `lunch_pipeline`, which makes us lunch. The 'verbs', `defrost` and `eat`, are Python functions and the inputs/outputs are food at different points of the process (`frozen`, `thawed` and `food`).
+
+```python
+cook_pipeline = pipeline(
+    [
+        node(func=defrost, inputs="frozen_veg", outputs="veg"),
+        node(func=grill, inputs="veg", outputs="grilled_veg"),
+    ]
+)
+
+lunch_pipeline = pipeline([node(func=eat, inputs="food", outputs=None)])
+
+cook_pipeline + lunch_pipeline
+```
+
+This combination will visualise since it's valid pre-runtime, but it will not run since `food` is not an output of the `cook_pipeline` because the output of the `cook_pipeline` is `grilled_veg`:
+
+![disjoined](../meta/images/cook_disjointed.png)
+
+* Combining `cook_pipeline + lunch_pipeline` will not work since `food` doesn't exist as an output of the `cook_pipeline`.
+* In this case, we will need to map `grilled_veg` to the expected input of `food`.
+
+The wrapper allows us to provide a mapping and fix this disconnect.
+
+```python
+from kedro.pipeline.modular_pipeline import pipeline
+
+prep_pipeline = pipeline(pipe=cook_pipeline, outputs={"grilled_veg": "food"})
+
+meal_pipeline = prep_pipeline + lunch_pipeline
+```
+
+Providing this input/output override will join up the pipeline nicely:
+
+![joined](../meta/images/cook_joined.png)
+
+```{note}
+In this example we have used the `+` operator to join two pipelines. You can also use `sum()` or pass a list of pipelines to the `pipe` argument.
+```
+
+
+## Example: Using a modular pipeline multiple times
+
+Reusing pipelines for slightly different purposes can be a real accelerator for teams and organisations when they reach a certain scale. In the real world, one could imagine pipelines with responsibilities like profiling or feature engineering being reused within the same project or even across projects via [micro-packaging](micro_packaging.md).
+
+* In an ideal world, we would like to use the `cook_pipeline` twice as you would `defrost` and `grill` multiple meals beyond the `veg` currently hard-coded.
+* Namespaces allow you to instantiate the same pipeline multiple times and keep operations isolated.
+* Like one provides arguments to a class' constructor, you can provide overriding inputs/outputs/parameters to the `pipeline()` wrapper.
+
+```{note}
+The set of overriding inputs and outputs must be a subset of the reused pipeline's "free" inputs and outputs, respectively. A free input is an input that isn't generated by a node in the pipeline, while a free output is an output that isn't consumed by a node in the pipeline. {py:meth}`Pipeline.inputs() <kedro.pipeline.Pipeline.inputs>` and {py:meth}`Pipeline.outputs() <kedro.pipeline.Pipeline.outputs>` can be used to list a pipeline's free inputs and outputs, respectively.
+```
+
```

Comment:
Why do we need both cooking and data science example to demonstrate how namespaces work?

Reply:
I agree with that. We want to move away from the cooking example and shift to something more practical, based on our spaceflights-pandas starter. I previously kept the cooking example, but let's go ahead and remove it.